Fixed updating entires with plural form

dhm80 opened the pull request:
Plural form could not be saved when using `updateEntry` method (plural form was deleted).

Here is a fix for that.
- updated Parser::updateEntry()
- added Parser::updateEntries()
- refactored removing fuzzy from Parser::updateEntry() into Parser::removeFuzzyFlagForMsgId()
- added tests

Commit:
Fixed updating entires with plural form
- updated Parser::updateEntry()
- added Parser::updateEntries()
- refactored removing fuzzy from Parser::updateEntry() into Parser::removeFuzzyFlagForMsgId()
- added tests

MAXakaWIZARD commented on src/Parser.php:
```diff
@@ -439,22 +439,68 @@ public function setEntries(array $entries)
-        unset($flags[array_search('fuzzy', $flags, true)]);
-        $this->entriesAsArrays[$original]['flags'] = $flags;
+    /**
+     * Allows modifaction of all translations of an entry
```
There is a typo here

MAXakaWIZARD commented on src/Parser.php:
```diff
@@ -439,22 +439,68 @@ public function setEntries(array $entries)
     }
 
+    /**
+     * Allows modifaction of one translations of an entry
```
Typo here

MAXakaWIZARD commented on src/Parser.php:
```diff
@@ -439,22 +439,68 @@ public function setEntries(array $entries)
+            or !is_string($translation)
+            or !isset($this->entriesAsArrays[$msgid]['msgstr'][$positionMsgstr])
+        ) {
+            throw new \Exception('Entry could not update the given translation');
```
This message is confusing. I think it should be `Cannot update entry translation`. Also please use `||` instead of `or`.

MAXakaWIZARD commented:
@dhm80 Thanks for this PR. Have you tested if file saved with this code can be opened in `PoEdit` and correctly displayed there?

Commit:
Minor changes

dhm80 commented:
@MAXakaWIZARD No problem, glad to help.

I changed the exception messages, corrected the typo and used `||` instead of `or` in the methods. That should be it ;)

And: Yes, the generated .po file works like charm in poEdit. I just testet that. Did you have any problems?

MAXakaWIZARD commented:
@dhm80 Thanks!
